### PR TITLE
BERT soft label generator

### PR DESCRIPTION
**Pull request by cloneisme**

### CODE REVIEW

File: iclx/soft_label_generator/infer_bert.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+from iclx.soft_label_generator.datamodule.sst2 import SST2DataModule
+from iclx.soft_label_generator.datamodule.sst5 import SST5DataModule
+from iclx.soft_label_generator.datamodule.trec import TRECDataModule
+from iclx.soft_label_generator.datamodule.ag_news import AGNewsDataModule
+from iclx.soft_label_generator.datamodule.yelp import YelpDataModule
+
+
+def save_to_jsonl(data: List[Dict[str, Any]], output_path: str):
+    with open(output_path, 'w') as f:
```
Comment (yoonichoi) [inline iclx/soft_label_generator/infer_bert.py:26]: 여기 [output_path 설정하는 라인](https://github.com/nemodleo/attentionx-icl/pull/27/files#diff-1dd7365fe64942c1ef7b2628e9b1a584ac31ec2c80bb83bb65ac20be8ee58c6aR120)에서 `{dataset_name}`을 folder name으로 받는데 만약 이 dataset name folder가 존재하지 않으면 makedir해주는 라인 하나 추가해주면 좋을 것 같아요

File: scripts/infer_bert_trec.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+from iclx.soft_label_generator.infer_bert import infer
+
+
+if __name__ == "__main__":
+    infer(
+        checkpoint_path="",  # Add the path to the checkpoint
+        dataset_name="trec",
+        batch_size=512,
+        file_name="train_trec.jsonl"
```
Comment (yoonichoi) [inline scripts/infer_bert_trec.py]: naming convention 따라서 `{dataset}-train.jsonl`로 통일했으면합니다ㅎㅎ

File: iclx/soft_label_generator/infer_bert.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        attention_mask = batch['attention_mask']
+        labels = batch['labels']
+
+        with torch.no_grad():
+            outputs = model(input_ids, attention_mask=attention_mask)
+            probs = F.softmax(outputs.logits, dim=-1)
+            probs = probs.cpu().numpy().tolist()
+            labels = labels.cpu().numpy().tolist()
+            label_texts = [data_module.label_texts()[label] for label in labels]
+            data.extend([
```
Comment (yoonichoi) [inline iclx/soft_label_generator/infer_bert.py:127]: `pseudo_gt` field 추가가 필요해보입니다!

File: Makefile [hunk 4]
```diff
@@ -10,15 +10,66 @@ poetry-install:
...
+vessl-run-sst2:
+	vessl run create -f vessl_exp/sst2.yaml
+
+vessl-run-sst2_gpt_j_6B:
+	vessl run create -f vessl_exp/sst2_gpt_j_6B.yaml
+
+vessl-run-sst2_gpt_neo_2.7B:
+	vessl run create -f vessl_exp/sst2_gpt_neo_2.7B.yaml
 
-run-distill:
```
Comment (yoonichoi) [inline Makefile]: 오잉 이부분 없어지면 안됩니당

File: Makefile [hunk 5]
```diff
@@ -10,15 +10,66 @@ poetry-install:
...
+	$(AUTO_POETRY) python scripts/bert/create_traindata_subj.py
+
+run-trec:
+	$(AUTO_POETRY) python scripts/trec.py
+
 vessl-workspace-init:
 	mkdir /root/.cache
 	ln -s /opt/.cache/huggingface /root/.cache/huggingface
 
-run-create_train:
```
Comment (yoonichoi) [inline Makefile]: 이거도 없어지면안되구요

File: Makefile [hunk 6]
```diff
@@ -10,15 +10,66 @@ poetry-install:
...
-	$(AUTO_POETRY) python scripts/create_train_with_pseudo.py $(SETUP_DICT)
+vessl-run-sst2:
```
Comment (yoonichoi) [inline Makefile]: 오히려 이 vessl-run-[모델] 이 라인들이 outdated된걸로 알고있습니다 (현재 develop branch에 있는 Makefile에서 민찬님이 추가하신 train/infer-bert 라인만 추가해주시면될것같습니다

File: iclx/inferencer/ppl_inferencer.py [hunk 7]
```diff
@@ -47,8 +47,7 @@ def inference(self,
                   ice_template: Optional[PromptTemplate] = None,
                   prompt_template: Optional[PromptTemplate] = None,
                   output_json_filepath: Optional[str] = None,
-                  output_json_filename: Optional[str] = None,
-                  pseudo_gt: Optional[str] = None) -> List:
```
Comment (yoonichoi) [inline iclx/inferencer/ppl_inferencer.py]: 헉 이것도 없어지면 안됩니다..

File: iclx/inferencer/ppl_inferencer.py [hunk 8]
```diff
@@ -72,7 +71,7 @@ def inference(self,
 
         # 4. Generate in-context examples for testing inputs
         for idx in range(len(ice_idx_list)):
-            ice.append(retriever.generate_ice(ice_idx_list[idx], ice_template=ice_template, pseudo_gt=pseudo_gt))
```
Comment (yoonichoi) [inline iclx/inferencer/ppl_inferencer.py]: 여기도 살리기 (diff가 잘못따진느낌?)

File: iclx/retriever/base_retriever.py [hunk 9]
```diff
@@ -81,7 +81,7 @@ def get_labels(self, ice_template: Optional[PromptTemplate] = None, prompt_templ
             labels = list(set(self.test_ds[self.dataset_reader.output_column]))
         return labels
 
-    def generate_ice(self, idx_list: List[int], ice_template: Optional[PromptTemplate] = None, pseudo_gt: Optional[str] = None) -> str:
```
Comment (yoonichoi) [inline iclx/retriever/base_retriever.py]: 여기도 살리기

File: iclx/utils/prompt_template.py
Comment (yoonichoi) [inline iclx/utils/prompt_template.py:1]: 이 파일 delete돼있는애들도 추가해야합니다!

**Review by yoonichoi — CHANGES_REQUESTED** [review]
공통적으로 `iclx`내의 파일들 diff가 잘못따진것같네요. 삭제된 라인들 복구시켜주시면 될 것 같습니다!

**Review by yoonichoi — APPROVED** [review]
